acai-ios/SectionControllers/OrderCustomizationListSectionController.swift

```diff
@@ -0,0 +1,103 @@
+//
+//  OrderCustomizationListSectionController.swift
+//  acai-ios
+//
+//  Created by Artesia Ko on 3/15/19.
+//  Copyright © 2019 Cornell AppDev. All rights reserved.
+//
+
+import UIKit
+import IGListKit
+
+class OrderCustomizationListSectionController: ListSectionController {
+    
+    // MARK: Delegates
+    weak var selectOptionDelegate: DidSelectOption?
+    
+    // MARK: Data
+    var customizationOptions: OrderCustomizationOptions!
+    var selectedBaseIndex: Int = 0
+    
+    // MARK: Constraint Constants
+    let baseCellHeightConstraint: CGFloat = 87
+    let toppingCellHeightConstraint: CGFloat = 55
+    
+    init(options: OrderCustomizationOptions) {
+        customizationOptions = options
+    }
+    
+    override func numberOfItems() -> Int {
+        return customizationOptions.options.count
+    }
+    
+    override func sizeForItem(at index: Int) -> CGSize {
+        guard let context = collectionContext else {
+            return .zero
+        }
+        let width = context.containerSize.width
+        switch customizationOptions.options[index].type {
+        case .base?:
+            return CGSize(width: width, height: baseCellHeightConstraint)
+        case .topping?:
+            return CGSize(width: width, height: toppingCellHeightConstraint)
+        case .none:
+            return .zero
+        }
+    }
+    
+    override func cellForItem(at index: Int) -> UICollectionViewCell {
+        var cellType: AnyClass = UICollectionViewCell.self
+        let option = customizationOptions.options[index]
+        switch option.type {
+        case .base?:
+            cellType = RadioSelectionCollectionViewCell.self
+            if option.isSelected {
+                selectedBaseIndex = index
+            }
+        case .topping?:
+            cellType = MultiSelectionCollectionViewCell.self
+        case .none:
+            cellType = UICollectionViewCell.self
+        }
+        
+        let cell = collectionContext!.dequeueReusableCell(of: cellType, for: self, at: index)
+        if let cell = cell as? RadioSelectionCollectionViewCell {
+            cell.customizationOption = option
+            cell.setCustomizationOptionLabelText()
+            cell.updateRadioSelection()
+        } else if let cell = cell as? MultiSelectionCollectionViewCell {
+            cell.customizationOption = option
+            cell.setOrderCustomizationOptionTitleLabelText()
+            cell.updateSelectionRect()
+        }
+        return cell
+    }
+    
+    override func didUpdate(to object: Any) {
+        guard let object = object as? OrderCustomizationOptions else { return }
+        customizationOptions = object
```

Comment:
do `customizationOptions = object as? OrderCustomizationOptions` and `var customizationOptions: OrderCustomizationOptions` instead of declaring it with `!`

Reply:
Value of optional type 'OrderCustomizationOptions?' must be unwrapped to a value of type 'OrderCustomizationOptions'

Reply:
wait what if you keep the `!`?